memo_app.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  'ページが見つかりません!'
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def save_memos(indicated_data)
+  File.write(MEMOS_FILE, JSON.pretty_generate(indicated_data))
+end
+
+def find_memo(existing_data)
+  existing_data.find { |memo| memo[:id] == params[:id] }
+end
+
+def assign_id
+  SecureRandom.uuid
+end
+
+get '/' do
+  @memos = load_memos
+  erb :index
+end
+
+get '/new' do
+  erb :new
+end
+
+post '/create' do
+  existing_data = load_memos
+  existing_data << { id: assign_id, title: params[:title], content: params[:content], created_at: Time.now, updated_at: Time.now }
+  save_memos(existing_data)
+  redirect '/'
+end
+
+get '/:id' do
+  existing_data = load_memos
+  @memo = find_memo(existing_data)
```

Comment:
```suggestion
  @memo = find_memo(params[:id])
```

idを渡せばデータをポンと返してくれるメソッドを定義してみましょう

Reply:
やってみました！